workers/googlePayments.js

```diff
@@ -0,0 +1,177 @@
+let nconf = require('nconf');
+let moment = require('moment');
+let request = require('request');
+let async = require('async');
+let iap = require('in-app-purchase');
+
+// Defined later
+let db;
+let queue;
+let habitrpgUsers;
+/* eslint-disable camelcase */
+
+// TODO fetch from api
+let subscriptionBlocks = {
+  basic_earned: {months: 1, price: 5},
+  basic_3mo: {months: 3, price: 15},
+  basic_6mo: {months: 6, price: 30},
+  google_6mo: {months: 6, price: 24, discount: true, original: 30},
+  basic_12mo: {months: 12, price: 48},
+};
+/* eslint-enable camelcase */
+
+iap.config({
+  // This is the path to the directory containing iap-sanbox/iap-live files
+  googlePublicKeyPath: nconf.get('IAP_GOOGLE_KEYDIR'),
+  googleAccToken: nconf.get('PLAY_API_ACCESS_TOKEN'),
+  googleRefToken: nconf.get('PLAY_API_REFRESH_TOKEN'),
+  googleClientID: nconf.get('PLAY_API_CLIENT_ID'),
+  googleClientSecret: nconf.get('PLAY_API_CLIENT_SECRET'),
+});
+
+function processUser (user, cb, jobStartDate, nextScheduledCheck) {
+  try {
```

Comment:
these functions should use promises, not callbacks, see the onboarding worker for an example